src/Microsoft.Framework.Runtime/DefaultHost.cs

```diff
@@ -58,6 +58,16 @@ public Assembly GetEntryPoint(string applicationName)
 
             Initialize();
 
+            // If the main project cannot be resolved, it means the app doesn't support current target framework
+            // (i.e. project.json doesn't contain a framework that is compatible to target framework of current runtime)
+            var mainProject = _applicationHostContext.DependencyWalker.Libraries
+                .Single(l => string.Equals(Project.Name, l.Identity.Name));
+            if (!mainProject.Resolved)
+            {
+                throw new InvalidOperationException(
+                    $"'{Project.Name}' doesn't support current runtime target framework '{_targetFramework}'");
```

Comment:
This error also isn't good enough. We need to tell people what runnable target frameworks are in their project.json and how they can potentially fix it. We can also use the IRuntimeEnvironment to show what DNX was invoked and tell the user which one should be invoked.

Something like:

```
'Project' doesn't support the current runtime target framework 'DNX Version=4.5.1'. 
```

/cc @Eilon @tdykstra here for help on the error message.